internal/p2p/uploads.go

```diff
@@ -2,142 +2,223 @@ package p2p
-							if appconfInstance.FFSkipVT {
-								result = true
-							} else {
-								result, err = psfs.GetVirusTotalWSVerdictByHash(fsha256) // true == safe
-								if err != nil {
-									fmt.Println("[ERROR] (GetVirusTotalVerdictByHash) " + string(err.Error()))
-									return
-								}
-							}
-						}
-
-						// fmt.Println("[INFOSEC] File Security check passed for file: " + f + " with SHA256: " + fsha256)
-						fresult = result
-					}
-
-				}
-			}
+	var count int64
+	var wg sync.WaitGroup
+	semaphore := make(chan struct{}, maxConcurrentUploads)
 
-			if fresult {
-				fcid := psfs.AddFileIPFS(folderPath + "/" + f)
-				if fcid != "" {
-					fmt.Println("[INFO] File: " + f + " ++added to IPFS with CID: " + fcid)
-					fileExtension, err := psfs.GetExtension(f)
-					if err != nil {
-						return
-					}
-
-					metadata := store.BaseMetadata{
-						FileSHA256: strings.ToLower(fsha256),
-						IPFSCID:    strings.ToLower(fcid),
-						FileType:   strings.ToLower(fileExtension),
-					}
-
-					errgs := store.GlobalStore.AddFile(metadata)
-					if errgs != nil {
-						fmt.Printf("[ERROR] failed to add file to GlobalStore: %w \n", errgs)
-						return
-					}
-					fmt.Println("[INFO] File: " + f + " ++added to GlobalStore with CID: " + fcid)
-					count = count + 1
-					if appconfInstance.FFMoveUpload {
-						err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.CacheFolder+"/"+f)
-						if err != nil {
-							fmt.Println("[ERROR] Error moving file: ", err)
-						}
-					}
-				}
-			} else {
-				if appconfInstance.FFSendFileVT {
-					// This was really to catch unknow files on VT
-
-					// TODO: 				if appconfInstance.SecurityCapability [1 2 3 4]
-
-					if appconfInstance.SecurityCapability == 4 {
-						fmt.Println("[INFO] Submitting File to 3rd Party for Security check for file: " + f + " with SHA256: " + fsha256)
-						submitresult, err := psfs.SendFileToVirusTotalWS(folderPath + "/" + f)
-						if err != nil {
-							fmt.Println("[ERROR] Error submitting file for security check: ", err)
-						}
-						if submitresult {
-							fmt.Println("[INFO] Submission Passed Security check for file: " + f + " with SHA256: " + fsha256)
-						} else {
-							fmt.Println("[ERROR] File Security check failed for file: " + f + " with SHA256: " + fsha256)
-							if appconfInstance.FFMoveUpload {
-								err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.RejectFolder+"/"+f)
-								if err != nil {
-									fmt.Println("[ERROR] Error moving file: ", err)
-								}
-							}
-						}
-					}
-				} else {
-					fmt.Println("[ERROR] File Security check failed for file: " + f + " with SHA256: " + fsha256)
-				}
-			}
-		} else {
-			fmt.Println("[ERROR] File type invalid for file: " + f)
-			if appconfInstance.FFMoveUpload {
-				err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.RejectFolder+"/"+f)
-				if err != nil {
-					fmt.Println("[ERROR] Error moving file: ", err)
-				}
+	for _, f := range files {
```

Comment:
If `f` is truly a filename as indicated by the goroutine below, then let's rename `f` to `filename` and `filles` to `filenames` here.